### ISSUE TITLE
Inconsistent order of levels, especially with horizontal boxplots

### ISSUE THREAD

**Issue by jawhiteley** [ggobi/ggally#229]
First of all, I was SUPER excited to find an implementation of pairplots in ggplot2, so THANK YOU for that.

I have noticed, however, that with a mix of categorical and continuous variables, the levels of the categorical variables are ordered inconsistently. This is most apparent with the 'boxplot' geom, which appears to be flipped using `coord_flip()`, and the levels end up in the opposite order of other labels (such as bar plots) in the same row.

Here is a minimal reproducible example to illustrate the issue:
```
library(GGally)
Data <- data.frame(var.cat = sample(LETTERS[1:3], 100, replace = TRUE, prob = c(0.2, 0.3, 0.5)), 
                   var.num = rlnorm(100),
                   var2 = sample(LETTERS[4:6], 100, replace = TRUE)
                   )
Data <- transform(Data, var.num = var.num * as.numeric(var.cat))   # values should be largest for var.cat == 'C'
ggpairs(Data)                # Image 1: It appears that values are largest for var.cat == 'A' :( 
ggpairs(Data[, 3:1])         # Image 2: The corresponding plot appears as expected.
```
![image](https://cloud.githubusercontent.com/assets/1281318/26219671/d3a95418-3bdd-11e7-9d1d-d9d156df5c6e.png)
![image](https://cloud.githubusercontent.com/assets/1281318/26219691/e0c9d0c8-3bdd-11e7-8377-51f6cf60a15a.png)

I have observed this issue in GGally v 1.3.0, running in R v 3.4.0; i386-w64-mingw32/i386 (32-bit). Is there a workaround to adjust the order of the levels?
I also note that factor levels only appear on the marginal labels if the other variable at that margin is also categorical. Based on the example above:
```
ggpairs(Data[, 1:2])         # No categorical labels :(
```
![image](https://cloud.githubusercontent.com/assets/1281318/26219727/057cbf66-3bde-11e7-8c18-b497e279f8bb.png)

**Comment by schloerke** [comment]
Hi Jonathon,

Glad you like `ggduo()`! 

Two things:
1. There was an addition in the latest GGally that changed the default combo section to `'box_no_facet'` instead of `'box'`.  This removed the facets that you are looking for.  To revert to box plot that uses facets...
```r
ggpairs(Data, upper = list(combo = 'box'))
```
2. I've submitted an issue to ggplot2 that they are not listening to the breaks argument. https://github.com/tidyverse/ggplot2/issues/2137 .  Unfortunately this is a ggplot2 error. :-/

By using 'box' instead of 'box_no_facet' for the upper combinations, both of your issues can be fixed immediately.  


As for some reasoning behind why the change to 'box_no_facet' was made... I really don't like how the facets push the plot inward.  The small white gap to help distinguish the groups is great, but the shifting is not good for data alignment.  If I could, I would have no faceting (or facets without labels) within a plot location for perfect alignment.  

As a side note... if you are looking to adding a legend, you could use the legend parameter if a legend exists within a plot.  Ex:

```r
# display the legend from the first row, first column plot
ggpairs(Data, aes(color = var.cat), legend = c(1,1), upper = list(combo = "box")))
```
<img width="724" alt="screen shot 2017-05-19 at 3 40 45 pm" src="https://cloud.githubusercontent.com/assets/93231/26264113/a9116eb2-3ca9-11e7-9a3f-faf5f8005ec3.png">


```r
# remove strips for better alignment
ggpairs(Data, aes(color = var.cat), legend = c(1,1), showStrips=FALSE, upper = list(combo = "box")))
```
<img width="752" alt="screen shot 2017-05-19 at 3 40 36 pm" src="https://cloud.githubusercontent.com/assets/93231/26264105/a5d55934-3ca9-11e7-84a6-61fd216c84dc.png">

**Comment by schloerke** [comment]
I'm going to close this issue for now as it is a ggplot2 issue and not something related to GGally.

- Barret

**Comment by schloerke** [comment]
This is related to GGally now

**Comment by schloerke** [comment]
This is fixed in #239 .  Closing as a release will be soon

**Comment by schloerke** [comment]
on CRAN 1.3.2
